src/exec.js

```diff
@@ -104,8 +105,11 @@ function execSync(cmd, opts, pipe) {
     code = parseInt(fs.readFileSync(codeFile, 'utf8'), 10);
   }
 
-  var stdout = fs.readFileSync(stdoutFile, 'utf8');
-  var stderr = fs.readFileSync(stderrFile, 'utf8');
+  // fs.readFileSync uses buffer encoding by default, so set
+  // the encoding only if opts.encoding wasn't set to 'buffer'
+  var encoding = opts.encoding !== 'buffer' ? opts.encoding : null;
```

Comment:
What happens if we get rid of the ternary and replace with `encoding = opts.encoding`? It seems weird to specify `'buffer'` if what it actually turns into is `null`